crates/oxc_language_server/src/main.rs

```diff
@@ -180,8 +188,38 @@ impl LanguageServer for Backend {
         }
     }
 
-    async fn did_change_watched_files(&self, _params: DidChangeWatchedFilesParams) {
+    async fn did_change_watched_files(&self, params: DidChangeWatchedFilesParams) {
         debug!("watched file did change");
+        if self.options.lock().await.use_nested_configs {
```

Comment:
If using nested configs, check the watched files to determine if they are `.oxlintrc.json` files (which are the only nested config files that are supported to my knowledge).

This just keeps the nested config map up to date with any workspace changes.